integration-tests/bulkExport.test.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import axios from 'axios';
+import { groupBy, mapValues, uniq, isEqual } from 'lodash';
+import createBundle from './createPatientPractitionerEncounterBundle.json';
+import BulkExportTestHelper, { ExportStatusOutput } from './bulkExportTestHelper';
+
+const FIVE_MINUTES_IN_MS = 5 * 60 * 1000;
+const { API_URL, API_ACCESS_TOKEN, API_KEY } = process.env;
+
+describe('Bulk Export', () => {
+    let bulkExportTestHelper: BulkExportTestHelper;
+
+    beforeAll(() => {
+        if (API_URL === undefined) {
+            throw new Error('API_URL environment variable is not defined');
+        }
+        if (API_ACCESS_TOKEN === undefined) {
+            throw new Error('API_ACCESS_TOKEN environment variable is not defined');
+        }
+        if (API_KEY === undefined) {
+            throw new Error('API_KEY environment variable is not defined');
+        }
+
+        const fhirUserAxios = axios.create({
+            headers: {
+                'x-api-key': API_KEY,
+                Authorization: `Bearer ${API_ACCESS_TOKEN}`,
+            },
+            baseURL: API_URL,
+        });
+
+        bulkExportTestHelper = new BulkExportTestHelper(fhirUserAxios);
+    });
+
+    const getResources = (bundleResponse: any): Record<string, any> => {
+        const resourceTypeToResource: any = {};
+        createBundle.entry.forEach((entry: any) => {
+            resourceTypeToResource[entry.resource.resourceType] = entry.resource;
+        });
+
+        const resourceTypeToExpectedResource: Record<string, any> = {};
+        bundleResponse.entry.forEach((entry: any) => {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const [location, resourceType, id] = entry.response.location.match(/(\w+)\/(.+)/);
+            const res = resourceTypeToResource[resourceType];
+            res.id = id;
+            res.meta = {
+                lastUpdated: entry.response.lastModified,
+                versionId: entry.response.etag,
+            };
+            resourceTypeToExpectedResource[resourceType] = res;
+        });
+
+        return resourceTypeToExpectedResource;
+    };
```

Comment:
L50 is mutating the Bundle object that comes from the JSON file. That Bundle is shared by all tests and should NOT be mutated. Looks like that's not causing any trouble now but it's brittle.

Also, I think that this function could be simplified. I think that we could use the fact that the bundle response preserves the order of the resources. Just loop through the response and copy the id to the bundle item with the same index.

The way it's written now is a bit hard to follow and breaks if there are more than one resource with the same type

Reply:
Good call about L50, I added `cloneDeep` from `lodash` so the modifications to `createBundle` is to a local copy, and not globally.

I simplified the method to use the `bundleResponse` order in creating the new `resourceTypeToExpectedResource` variable. And yep, you're right about how this data structure only supports one value for each `resourceType`. I wrote the test this way to keep the test simpler. We could make value of the `resourceTypeToExpectedResource` object be an array of resources. However, the test doesn't require us checking for multiple resources per `resourceType`, therefore an array is not necessary.